src/lib/coda_transition/external_transition.ml

```diff
@@ -887,18 +883,14 @@ module Staged_ledger_validation = struct
       Protocol_state.blockchain_state (protocol_state transition)
     in
     let staged_ledger_diff = staged_ledger_diff transition in
-    let diff_state_body_hash =
-      Staged_ledger_diff.state_body_hash staged_ledger_diff
-    in
-    let parent_state_body_hash =
-      Protocol_state.(Body.hash @@ body parent_protocol_state)
-    in
     let%bind ( `Hash_after_applying staged_ledger_hash
              , `Ledger_proof proof_opt
              , `Staged_ledger transitioned_staged_ledger
              , `Pending_coinbase_data _ ) =
       Staged_ledger.apply ~logger ~verifier parent_staged_ledger
         staged_ledger_diff
+        ~state_body_hash:
+          Protocol_state.(Body.hash @@ body parent_protocol_state)
```

Comment:
This wasn't introduced by your PR (so it doesn't need to be fixed here), but we should update the code so that we don't need to recompute the body hashes of protocol states.

Reply:
I created issue #4139 to capture this work.